### PR TITLE
fix: 修复合并单元格临界情况被错误移除的问题

### LINKED ISSUE #1570: 🐛 合并单元格patialVisible判断问题
### 🏷 Version

| Package      | Version |
| -------------- | --------- |
| @antv/s2       |          |
| @antv/s2-react |          |
| @antv/s2-vue   |          |

### Sheet Type
合并单元格在滚动式，当恰好只有一行显示时，visible的判断有问题，此时mergeCell意外被删除，原本的DataCell的出现，导致页面出现闪动：

- [ ] PivotSheet
- [ ] TableSheet
- [ ] GridAnalysisSheet
- [ ] StrategySheet

### PR DESCRIPTION
**Pull request by wjgogogo**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] Solve the issue and close #1570 

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
|   |   |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/__tests__/unit/utils/merge-cell-spec.ts [hunk 1]
```diff
@@ -372,21 +371,5 @@ describe('Merge Cells Test', () => {
 
       expect(result).toEqual([{ viewMeta: { id: '2' } }]);
     });
-
-    test('should get TempMergedCells when MergedCell isPartiallyVisible is true. (differenceTempMergedCells)', () => {
```
Comment (wjgogogo) [inline packages/s2-core/__tests__/unit/utils/merge-cell-spec.ts:376]: 不再存在这样的场景

**Review by stone-lyl — APPROVED** [review]
先删除吧，后续再看看有没有其他问题

**Pull request merged**